pyiron_contrib/atomistics/runner/job.py

```diff
@@ -1,235 +1,444 @@
+
+
+class PotentialFittingBase(GenericJob):
+    """FIXME: Class copied from AtomicRex in pyiron_contrib."""
+
+    def __init__(self, project, job_name):
+        """Initialize the class.
+
+        Args:
+            project (FIXME): The project container in which the job is created.
+            job_name (str):  The label of the job (used for all directories).
+        """
+        super().__init__(project, job_name)
+
+
+class RunnerInput(DataContainer):
+    """Extend the DataContainer class for future customization."""
 
-__author__ = "Marvin Poul"
-__copyright__ = "Copyright 2020, Max-Planck-Institut für Eisenforschung GmbH - " \
-                "Computational Materials Design (CM) Department"
-__version__ = "0.1"
-__maintainer__ = "Marvin Poul"
-__email__ = "poul@mpie.de"
-__status__ = "development"
-__date__ = "March 3, 2021"
+    __version__ = '0.3.0'
+    __hdf_version__ = '0.3.0'
 
-AngstromToBohr = 1.88972612
-ElectronVoltToHartree = 0.03674932218
+    def __init__(self, *args, **kwargs):
+        """Initialize the class."""
+        super(RunnerInput, self).__init__(*args, **kwargs)
 
 
-class RunnerFit(GenericJob):
+class RunnerStructureContainer(StructureStorage):
+    """Store chemical structures as a Runner training dataset."""
+
+    __version__ = '0.3.0'
+    __hdf_version__ = '0.3.0'
+
     def __init__(self, project, job_name):
-        super().__init__(project, job_name) 
-        self._training_ids = []
+        """Initialize the class.
+
+        Args:
+            project (FIXME): The project container in which the job is created.
+            job_name (str):  The label of the job (used for all directories).
+        """
+        super(RunnerStructureContainer, self).__init__()
+
+    def append(self, structure):
+        """Append `structure` to the class storage."""
+        # If the user appends an ASE Atoms object, try to obtain energies,
+        # forces, etc. and store them.
+        if isinstance(structure, Atoms):
+            struct = ase_to_pyiron(structure)
+            dft_charges = structure.get_initial_charges()
+
+            if structure.calc:
+                dft_energy = structure.calc.get_potential_energy()
+                dft_forces = structure.calc.get_forces()
+
+                # Only RunnerSinglePointCalculator object's store a separate
+                # total charge property. For all other calculators, sum up
+                # the atomic charges.
+                if isinstance(structure.calc, RunnerSinglePointCalculator):
+                    totalcharge = structure.calc.results['totalcharge']
+                else:
+                    totalcharge = np.sum(dft_charges)
+        else:
+            struct = structure
+
+        self.add_structure(
+            struct,
+            dft_energy=dft_energy,
+            dft_forces=dft_forces,
+            dft_charges=dft_charges,
+            totalcharge=totalcharge
+        )
+        return struct
+
+    def to_ase(self):
```

Comment:
Unfortunately, I do need this at the moment, because the ASE calculator for RuNNer requires the total charge as a calculator property to function correctly. This is only handled through the `RunnerSinglepointCalculator` class right now. However, I agree that is tedious and will kick off a discussion with the ASE people once we have tested the ASE calculator a little bit more within our group.

Reply:
Fair let's leave it in then for now.